src/machines/java/com/enderio/machines/common/item/FluidTankItem.java

```diff
@@ -26,4 +49,63 @@ public BlockEntityWithoutLevelRenderer getCustomRenderer() {
             }
         });
     }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level pLevel, List<Component> tooltip, TooltipFlag pFlag) {
+        ItemTooltip.addShiftKeyMessage(tooltip);
+        Optional<IFluidHandlerItem> fluidHandler = stack.getCapability(ForgeCapabilities.FLUID_HANDLER_ITEM).resolve();
+
+        ItemTooltip.addFluidTankMessage(tooltip, fluidHandler.get().getFluidInTank(0));//always present
+        super.appendHoverText(stack, pLevel, tooltip, pFlag);
+    }
+
+    @Override
+    public @Nullable ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt) {
+        return new FluidItemStack(stack, capacity);
+    }
+
+
+    public class FluidItemStack extends FluidHandlerItemStack{
+        private static final String BLOCK_ENTITY_TAG = "BlockEntityTag";
+        /**
+         * @param container The container itemStack, data is stored on it directly as NBT.
+         * @param capacity  The maximum capacity of this fluid tank.
+         */
+        public FluidItemStack(@NotNull ItemStack container, int capacity) {
+            super(container, capacity);
+        }
+
+        @NotNull
+        public FluidStack getFluid() {
+            CompoundTag tagCompound = container.getTag();
+            if(tagCompound == null) return FluidStack.EMPTY;
+            CompoundTag BETag = tagCompound.getCompound(BLOCK_ENTITY_TAG);
+            if (!BETag.contains("fluid")) return FluidStack.EMPTY;
+            else return FluidStack.loadFluidStackFromNBT(BETag.getCompound("fluid"));
```

Comment:
maybe use multiple Optional.map chains or fix the early returns like above

Reply:
Okay, so I don't understand how to do it, but chatGPT did apparently, so done now...